mongodbatlas/resource_mongodbatlas_online_archive.go

```diff
@@ -0,0 +1,414 @@
+					},
+					"field_type": {
+						Type:     schema.TypeString,
+						Computed: true,
+					},
+				},
+			},
+		},
+		// mongodb_atlas id
+		"atlas_id": {
+			Type:     schema.TypeString,
+			Computed: true,
+		},
+		"paused": {
+			Type:     schema.TypeBool,
+			Optional: true,
+			Computed: true,
+		},
+		"state": {
+			Type:     schema.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func resourceMongoDBAtlasOnlineArchiveCreate(d *schema.ResourceData, meta interface{}) error {
+	// Get client connection
+	conn := meta.(*matlas.Client)
+	projectID := d.Get("project_id").(string)
+	clusterName := d.Get("cluster_name").(string)
+
+	inputRequest := mapToArchivePayload(d)
+	outputRequest, _, err := conn.OnlineArchives.Create(context.Background(), projectID, clusterName, &inputRequest)
+
+	if err != nil {
+		return fmt.Errorf(errorOnlineArchivesCreate, err)
+	}
+
+	d.SetId(encodeStateID(map[string]string{
+		"project_id":   projectID,
+		"cluster_name": outputRequest.ClusterName,
+		"atlas_id":     outputRequest.ID,
+	}))
+
+	return resourceMongoDBAtlasOnlineArchiveRead(d, meta)
+}
+
+func resourceMongoDBAtlasOnlineArchiveRead(d *schema.ResourceData, meta interface{}) error {
+	// getting the atlas id
+	conn := meta.(*matlas.Client)
+	ids := decodeStateID(d.Id())
+
+	atlasID := ids["atlas_id"]
+	projectID := ids["project_id"]
+	clusterName := ids["cluster_name"]
+
+	outOnlineArchive, _, err := conn.OnlineArchives.Get(context.Background(), projectID, clusterName, atlasID)
+
+	if err != nil {
+		reset := strings.Contains(err.Error(), "404") && !d.IsNewResource()
+		if reset {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("error MongoDBAtlas Online Archive with id %s, read error %s", atlasID, err.Error())
+	}
+
+	newData := fromOnlineArchiveToMapInCreate(outOnlineArchive)
+
+	for key, val := range newData {
+		if err := d.Set(key, val); err != nil {
+			return fmt.Errorf("error MongoDBAtlas Online Archive with id %s, read error %s", atlasID, err.Error())
+		}
+	}
+	return nil
+}
+
+func resourceMongoDBAtlasOnlineArchiveDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*matlas.Client)
+	ids := decodeStateID(d.Id())
+	atlasID := ids["atlas_id"]
+	projectID := ids["project_id"]
+	clusterName := ids["cluster_name"]
+
+	_, err := conn.OnlineArchives.Delete(context.Background(), projectID, clusterName, atlasID)
+
+	if err != nil {
+		alreadyDeleted := strings.Contains(err.Error(), "404") && !d.IsNewResource()
+		if alreadyDeleted {
+			return nil
+		}
+
+		return fmt.Errorf(errorOnlineArchivesDelete, err, atlasID)
+	}
+	return nil
+}
+
+func resourceMongoDBAtlasOnlineArchiveImportState(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	conn := meta.(*matlas.Client)
+	parts := strings.Split(d.Id(), "-")
+
+	if len(parts) != 3 {
+		return nil, errors.New("import format error to import a Mongo Online Archive, use the format {project_id}-{cluste_rname}-{atlas_id} ")
```

Comment:
Mongo need the DB and Atlas : 
return nil, errors.New("import format error to import a Mongo**DB** **Atlas** Online Archive, use the format {project_id}-{cluste_rname}-{atlas_id} ")

Also this is where atlas id might get truly confusing for users.  It doesn't make one think archive Id.

Reply:
agree :) modified to be archive_id

Reply:
thank you!